happi/client.py

```diff
@@ -600,16 +603,27 @@ def _store(self, device, insert=False):
         post.update({'type': tpe,
                      'creation': creation,
                      'last_edit': ttime.ctime()})
+
         # Find id
         try:
+            # In case we want to update the name of an entry
+            # We want to add a new entry, and delete the old one
             _id = post[self._id_key]
+            if the_old_name and the_old_name != post[self._id_key]:
```

Comment:
I think you should move this block out of the `try/except KeyError`, as we're only trying to see if `post[id_key]` exists.

Reply:
Sounds good, moved it in 55a83ad